### DISCUSSION TITLE
Receiving return values

### DISCUSSION THREAD

**Discussion by eliezerben** [livewire/livewire#8119] [1. Help] [answered]
Hi,

I'm trying out this section of the documentation: https://livewire.laravel.com/docs/actions#receiving-return-values

> For even more power, invoked $wire actions return a promise while the network request is processing. When the server response is received, the promise resolves with the value returned by the backend action.

But this results in continuous network requests sent to the backend:
![msedge_tiyHDLHuKP](https://github.com/livewire/livewire/assets/19612629/a57380ec-95b2-459c-b92c-17a14cbfd245)

The payload of each request looks like this:
![Screenshot 2024-03-15 030524](https://github.com/livewire/livewire/assets/19612629/06030f90-0ef3-4094-8f40-c668ae49df91)

This is the code I'm trying out (Similar to the code mentioned in the docs):

**ShowPosts.php**
```
<?php

namespace App\Livewire;

use App\Models\Post;
use Livewire\Component;

class ShowPosts extends Component
{
    public function render()
    {
        return view('livewire.show-posts');
    }

    public function getPostCount()
    {
        return Post::count();
    }
}
```

**show-posts.blade.php**
```
<div>
    Post count: <span x-text="await $wire.getPostCount()"></span>
</div>
```

I feel like I'm missing something obvious. Please help.

**Comment by danie-ramdhani** [comment]
oh yeah you're right. it will make infinite network request.
maybe this will work:

```blade
<div x-data="{
    postCount: $wire.getPostCount(),
}">
    <span x-text="postCount"></span>
</div>
```

the best solution is using computed properties:

```blade
<div>
    Post count: <span>{{ $this->getPostCount }}</span>
</div>
```

```php
#[\Livewire\Attributes\Computed]
public function getPostCount()
{
    return Post::count();
}
```

**Reply by eliezerben** [reply]
> ```
> <div x-data="{
>     postCount: $wire.getPostCount(),
> }">
>     <span x-text="postCount"></span>
> </div>
> ```
This works. Thanks

But why does the original snippet result in infinite requests?

And why does the documentation suggest doing something which results in infinite requests?

**Reply by danie-ramdhani** [reply, marked as answer]
mmm yeah... idk, I've never use function with return value in `x-text` outside alpine `x-data` because we have computed properties...
maybe because it always evaluated when livewire update happened:
https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-text.js

```blade
<div wire:ignore> // this will prevent the evaluated
    Post count: <span x-text="await $wire.getPostCount()"></span>
</div>
```

if you really want to use `$wire` object to get `postCount`, you can use `x-init`

```blade
<div>
    Post count: <span x-init="$el.innerHTML = await $wire.getPostCount()"></span>
</div>
```

**Reply by danie-ramdhani** [reply]
made a PR: https://github.com/livewire/livewire/pull/8122

**Reply by eliezerben** [reply]
> mmm yeah... idk, I've never use function with return value in `x-text` outside alpine `x-data` because we have computed properties... maybe because it always evaluated when livewire update happened: https://github.com/alpinejs/alpine/blob/main/packages/alpinejs/src/directives/x-text.js

That makes sense for why it's happening. Thanks.
